src/kernel/arch/x86/syscalls.zig

```diff
@@ -0,0 +1,70 @@
+const arch = @import("arch.zig");
+const testing = @import("std").testing;
+const assert = @import("std").debug.assert;
+const isr = @import("isr.zig");
+const log = @import("../../log.zig");
+
+pub const INTERRUPT: u16 = 0x80;
+pub const NUM_HANDLERS = 256;
+
+/// A syscall handler
+pub const SyscallHandler = fn (ctx: *arch.InterruptContext) void;
+/// Errors that syscall utility functions can throw
+pub const SyscallError = error {
+    SyscallExists
+};
+
+var handlers = []?SyscallHandler{null} ** NUM_HANDLERS;
+
+fn handle(ctx: *arch.InterruptContext) void {
+    // The syscall number is put in eax
+    const syscall = ctx.eax;
+    if (syscall < NUM_HANDLERS) {
+        if (handlers[syscall]) |handler| {
+            handler(ctx);
+        } else {
+            log.logWarning("Syscall {} triggered but not registered\n", syscall);
+        }
+    }
+}
+
+///
+/// Register a syscall so it  can be called by triggering interrupt 128
+///
+/// Arguments:
+///     IN syscall: u8 - The syscall number to register the handler with
+///     IN handler: SyscallHandler - The handler to register the syscall with
+///
+pub fn registerSyscall(syscall: u8, handler: SyscallHandler) !void {
+    if (handlers[syscall]) |_|
+        return SyscallError.SyscallExists;
+    handlers[syscall] = handler;
+}
+
+///
+/// Trigger a syscall
+///
+/// Arguments:
+///     IN syscall: u8 - The syscall to trigger. Is put in eax and must have been registered.
+///
+pub fn doSyscall(syscall: u8) void {
+    asm volatile ("mov %%ecx, %%eax" :: [syscall] "{ecx}" (syscall));
+    asm volatile ("int $0x80");
```

Comment:
This can be done in one asm volatile statement. Also what if the syscall requires arguments. see: https://ziglang.org/documentation/0.4.0/#Assembly

Reply:
Good idea. Do you know how to pass constants as arguments to the assembly? It would nice to not have to hardcode the `0x80`.

Reply:
Haven't looked into this, but i do think u can.

Reply:
Regarding the arguments, either the syscall handler should pop off the arguments itself or we can add some functionality here once the requirements are clear, or maybe you think that should be added in this PR.

Reply:
I decided to add them